client/my-sites/domains/domain-management/list/domain-only-upsell-carousel.tsx

```diff
@@ -24,94 +57,192 @@ const DomainOnlyUpsellCarousel = ( props: DomainOnlyUpsellCarouselProps ): JSX.E
+									</PopoverMenuItem>
+									<PopoverMenuItem onClick={ () => hideCard( '1m', cardNoticeType, cardName ) }>
+										{ translate( 'For a month' ) }
+									</PopoverMenuItem>
+								</PopoverMenu>
+							) }
 						</div>
 					</div>
 				</div>
 				<TrackComponentView
-					eventName="calypso_domain_only_upsell_carousel_impression_add_email_card"
-					eventProperties={ { content_type: 'add-professional-email' } }
+					eventName={ eventTrackViewName }
+					eventProperties={ { content_type: cardName } }
 				/>
 			</Card>
 		);
 	};
 
-	const cards = [ renderAddSiteCard() ];
-	if ( ! hasPaidEmailWithUs( domain ) ) {
-		cards.push( renderEmailCard() );
+	if ( isRequestingDomainNotices ) return null;
+
+	const cards = [];
+
+	if ( ! hideCreateSiteCard ) {
+		cards.push(
+			renderCard( {
+				actionUrl: createSiteFromDomainOnly( domain.domain, domain.blogId ),
+				imageUrl: createSiteImage,
+				title: translate( 'Create a site for %(domain)s', {
+					args: { domain: domain.domain },
+				} ),
+				subtitle: translate( 'Choose a theme, customize and launch your site.' ),
+				ref: hideCreateSiteCardButtonRef,
+				buttonLabel: translate( 'Create site' ),
+				cardName: 'create-site',
+				cardNoticeType: 'upsellCarouselHideCreateSiteCardUntil',
+				eventTrackViewName: 'calypso_domain_only_upsell_carousel_impression_create_site_card',
+				areHideOptionsVisible: areHideSiteCardOptionsVisible,
+				setHideOptionsVisible: setHideSiteCardOptionsVisible,
+			} )
+		);
+	}
+
+	if ( ! hasPaidEmailWithUs( domain ) && ! hideAddEmailCard ) {
+		cards.push(
+			renderCard( {
+				actionUrl: emailManagementPurchaseNewEmailAccount( domain.domain, domain.domain ),
+				imageUrl: addEmailImage,
+				title: translate( 'Add email for %(domain)s', {
+					args: { domain: domain.domain },
+				} ),
+				subtitle: translate( 'Send and receive emails from %(email)s', {
+					args: { email: `youremail@${ domain.domain }` },
+				} ),
+				ref: hideAddEmailCardButtonRef,
+				buttonLabel: translate( 'Add professional email' ),
+				cardName: 'add-email',
+				cardNoticeType: 'upsellCarouselHideAddEmailCardUntil',
+				eventTrackViewName: 'calypso_domain_only_upsell_carousel_impression_add_email_card',
+				areHideOptionsVisible: areHideEmailCardOptionsVisible,
+				setHideOptionsVisible: setHideEmailCardOptionsVisible,
+			} )
+		);
 	}
 
+	if ( cards.length === 0 ) return null;
+
 	return (
-		<DotPager className="domain-only-upsell-carousel" hasDynamicHeight showControlLabels={ false }>
+		<DotPager
+			className="domain-only-upsell-carousel"
+			hasDynamicHeight={ false }
+			showControlLabels={ false }
+		>
 			{ cards }
 		</DotPager>
 	);
 };
 
-export default connect( null, { dispatchRecordTracksEvent: recordTracksEvent } )(
-	DomainOnlyUpsellCarousel
-);
+const shouldHideCard = ( date: string ): boolean => {
+	const reminderEnd = moment( new Date( date ) );
+	return reminderEnd.isValid() && moment().isBefore( reminderEnd );
+};
```

Comment:
I think this also needs a condition for when `date` is null, as `new Date( null )` returns the Unix epoch (1970-01-01) which might have unintended consequences. I'm not sure, but doesn't `moment( date )` work, without needing to transform `date` into a `Date` object?

Reply:
In case date is `null` it's probably not a problem (it acts like an expired reminder) but I refactored the code to improve readability.

I had to check `null` date because `moment(null)` return `null`

```
const shouldHideCard = ( date: string | null ): boolean => {
	if ( ! date ) return false;
	const reminderEnd = moment( date );
	return reminderEnd.isValid() && moment().isBefore( reminderEnd );
};

```